microservices/CFDP/app/models/cfdp_pdu.rb

```diff
@@ -159,14 +173,21 @@ def self.build_initial_pdu(type:, destination_entity:, file_size:, segmentation_
     else
       pdu.write("CRC_FLAG", "CRC_NOT_PRESENT")
     end
-    if file_size >= 4_294_967_296
+
+    # Version 0 doesn't support large files
+    if version == 0 or file_size < 4_294_967_296
+      pdu.write("LARGE_FILE_FLAG", "SMALL_FILE")  # Equals 0
+    else
       pdu.write("LARGE_FILE_FLAG", "LARGE_FILE")
+    end
+
+    if version == 0
+      pdu.write("SEGMENTATION_CONTROL", 0) # Always 0
     else
-      pdu.write("LARGE_FILE_FLAG", "SMALL_FILE")
+      pdu.write("SEGMENTATION_CONTROL", segmentation_control)
     end
-    pdu.write("SEGMENTATION_CONTROL", segmentation_control)
     pdu.write("ENTITY_ID_LENGTH", destination_entity['entity_id_length'])
-    pdu.write("SEGMENT_METADATA_FLAG", "NOT_PRESENT") # Not implemented
+    pdu.write("SEGMENT_METADATA_FLAG", "NOT_PRESENT") # Not implemented - always 0
```

Comment:
version 0 doesn't have the SEGMENT_METADATA_FLAG key if I look above at the keys array

Reply:
See above.